src/app.ts

```diff
@@ -224,22 +242,27 @@ export class App extends Construct {
 
   }
 
+  private toJsonInternal(chart: Chart): any[] {
```

Comment:
I would really love to avoid this, we intentionally made an effort to use the same `toJson` method everywhere. Can we somehow reuse the normal `toJson` and make `resolveDependencies(app)` and `validate(app)` only execute once? i.e - store a "validated" and "resolved" fields on the app instance and just no-op in case they are true?

Reply:
I'm also pretty sure a test is failing in the build now because of this...

Reply:
I'll have a look 👍🏻

Reply:
@iliapolo I have now removed the `toJsonInternal` and reverted that change. I had a look at implementing it in a clean way, but this is not possible without adding additional overrides to `chart.toJson` so that we can pass in the cache. 

I'm not feeling comfortable in adding an internal state to app that it has already been validated/resolved the dependencies, because of the reasons above (invalidating the cache if the chart is modified during `toSynth` calls). 
The current PR is already the majority of the performance boost. We should go with this PR and add the second change in a follow-up. Thoughts?

Reply:
Agreed